api/src/main/java/org/motechproject/nms/api/web/BaseController.java

```diff
@@ -36,62 +35,61 @@ protected static boolean validateFieldPresent(StringBuilder errors, String field
-    }
-
-    protected static boolean validateFieldNumeric15(StringBuilder errors, String fieldName, String value) {
-        return validateFieldNumericPattern(NUMERIC_PATTERN_15, errors, fieldName, value);
+    private static boolean validateField15Digits(StringBuilder errors, String fieldName, Long value) {
+        if (!validateFieldPresent(errors, fieldName, value)) {
+            return false;
+        }
+        if (value >= SMALLEST_15_DIGIT_NUMBER && value <= LARGEST_15_DIGIT_NUMBER) {
+            return true;
+        }
+        errors.append(String.format(INVALID, fieldName));
+        return false;
     }
 
-    protected static boolean validateFieldCallStatus(StringBuilder errors, String fieldName, String value) {
-        if (!validateFieldNumericPattern(NUMERIC_PATTERN, errors, fieldName, value)) {
+    protected static boolean validateFieldCallStatus(StringBuilder errors, String fieldName, Integer value) {
+        if (!validateFieldPresent(errors, fieldName, value)) {
             return false;
         }
-        if (CallStatus.isValid(Integer.parseInt(value))) {
+        if (CallStatus.isValid(value)) {
             return true;
         }
         errors.append(String.format(INVALID, fieldName));
         return false;
     }
 
-    protected static boolean validateFieldCallDisconnectReason(StringBuilder errors, String fieldName, String value) {
-        if (!validateFieldNumericPattern(NUMERIC_PATTERN, errors, fieldName, value)) {
+    protected static boolean validateFieldCallDisconnectReason(StringBuilder errors, String fieldName,
+                                                               Integer value) {
+        if (!validateFieldPresent(errors, fieldName, value)) {
             return false;
         }
-        if (CallDisconnectReason.isValid(Integer.parseInt(value))) {
+        if (CallDisconnectReason.isValid(value)) {
             return true;
         }
         errors.append(String.format(INVALID, fieldName));
         return false;
     }
 
-    protected StringBuilder validate(String callingNumber, String callId) {
+    protected StringBuilder validate(Long callingNumber, Long callId) {
         StringBuilder failureReasons = new StringBuilder();
 
-        validateFieldNumeric10(failureReasons, "callingNumber", callingNumber);
-        validateFieldNumeric15(failureReasons, "callId", callId);
+        validateField10Digits(failureReasons, "callingNumber", callingNumber);
+        validateField15Digits(failureReasons, "callId", callId);
 
         return failureReasons;
     }
 
-    protected StringBuilder validate(String callingNumber, String operator, String circle, String callId) {
+    protected StringBuilder validate(Long callingNumber, Long callId, String operator, String circle) {
         StringBuilder failureReasons = validate(callingNumber, callId);
 
         validateFieldPresent(failureReasons, "operator", operator);
```

Comment:
According to the spec operator and circle are a max of 255 characters, so we should probably check that.